### PR TITLE
aws public subnet explainability

### PR DESCRIPTION
**Pull request by haim-kermany**
implementation of the first bullet of:
https://github.com/np-guard/vpc-network-config-analyzer/issues/756

### CODE REVIEW

File: pkg/vpcmodel/explainabilityPrint.go [hunk 1]
```diff
@@ -206,6 +213,12 @@ func (g *groupedConnLine) explainPerCaseStr(c *VPCConfig, src, dst EndpointElem,
 	case crossVpcRouterRequired(src, dst) && crossVpcRouter != nil && crossVpcConnection.IsEmpty():
 		return fmt.Sprintf("%vAll connections will be blocked since transit gateway denies route from source to destination"+tripleNLVars,
 			noConnection, headerPlusPath, details)
+	case isAtPrivateSubnet(dst) && src.IsExternal():
```
Comment (ShiriMoran) [inline pkg/vpcmodel/explainabilityPrint.go:216]: The cases in which we provide a single line explaining why there is no connection should be minimized see #655. In this specific case we can provide full explainability and integrate the private subnet in the path and details. 

We could defer the explanation part of this PR to latter PRs - I can take responsibility to this part.

**Review by ShiriMoran — CHANGES_REQUESTED** [review]
1. Please add tests
2. See my comment regarding explanation

**Comment by haim-kermany** [comment]
> 1. Please add tests
> 
>     2. See my comment regarding explanation

tests will be added in a different PR
please follow issue https://github.com/np-guard/vpc-network-config-analyzer/issues/756

File: pkg/vpcmodel/explainabilityPrint.go (hunk 1 above)
Comment (haim-kermany) [reply pkg/vpcmodel/explainabilityPrint.go:216]: OK, will be resolved in another PR, by our explainability expert.
added a bullet to the relevant issue https://github.com/np-guard/vpc-network-config-analyzer/issues/756

**Review by adisos — APPROVED** [review]

File: pkg/vpcmodel/explainabilityPrint.go (hunk 1 above)
Comment (ShiriMoran) [reply pkg/vpcmodel/explainabilityPrint.go:216]: Perhaps better remove it all together

**Review by ShiriMoran — APPROVED** [review]

**Pull request merged**